### PR TITLE
[Yarn Rewrite] Added a script to separate the mappings before running gradle

### PR DESCRIPTION
**Pull request by BleachDrinker420**
Added yarn.py which separates the mappings, runs a gradle command then merges the mappings back.

The intermediary_helper code is pretty spaghetti but i've tested quite a few cases and everything seems to work fine. adding, renaming and deleting all works and fields, methods and classes with duplicated names also works fine but i'd still recommend to have some people test it before merging.

Since i used the start file from #194 that PR isn't needed anymore and should be closed if this one gets merged.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -39,7 +39,7 @@ jobs:
           architecture: 'x64'
       - uses: actions/setup-java@v2
         with:
-          distribution: 'semeru' # See 'Supported distributions' for available options
+          distribution: 'temurin' # See 'Supported distributions' for available options
```
Comment (BluCobalt) [inline .github/workflows/build.yml:42]: Openj9 tho

File: yarn.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        version = args[1]
+        command = " ".join(args[2:])
+
+    os.environ["MC_VERSION"] = version
+    seperate(version)
+
+    if command.startswith("yarn"):
+        start_autosave()
+
+    os.system("gradlew " + command)
```
Comment (arthurbambou) [inline yarn.py:47]:
```suggestion
    os.system("./gradlew " + command)
```

**Review by arthurbambou — CHANGES_REQUESTED** [review]
wont work on windows tho

**Comment by arthurbambou** [comment]
It seems like some conflicts slipped through before :flushed: on 1.11.2
```
Mapping source name conflicts detected:
  net/minecraft/class_3106 METHOD method_13872 ((Lnet/minecraft/class_3146;)Lnet/minecraft/class_1653;) -> [net/minecraft/class_3106/method_5745, net/minecraft/class_3106/method_5757, net/minecraft/class_3109/method_5757]
  net/minecraft/class_1357 METHOD method_5816 ((Lnet/minecraft/class_982;)Lnet/minecraft/class_1653;) -> [net/minecraft/class_1357/method_5745, net/minecraft/class_1357/method_5757]
  net/minecraft/class_3109 METHOD method_13872 ((Lnet/minecraft/class_3146;)Lnet/minecraft/class_1653;) -> [net/minecraft/class_3106/method_5757, net/minecraft/class_3109/method_5745, net/minecraft/class_3109/method_5757]
  net/minecraft/class_3103 METHOD method_5816 ((Lnet/minecraft/class_982;)Lnet/minecraft/class_1653;) -> [net/minecraft/class_1357/method_5745, net/minecraft/class_3103/method_5757]
  net/minecraft/class_1352 METHOD method_13872 ((Lnet/minecraft/class_3146;)Lnet/minecraft/class_1653;) -> [net/minecraft/class_3106/method_5757, net/minecraft/class_3109/method_5757, net/minecraft/class_1352/method_5745]
```

**Comment by BleachDrinker420** [comment]
The conflicts seems to be fixed, weird that it only happens when the methods are unmapped

**Review by arthurbambou — APPROVED** [review]

**Pull request merged**